src/main/java/gregtech/api/capability/impl/FuelRecipeLogic.java

```diff
@@ -44,7 +44,7 @@ public FuelRecipeLogic(MetaTileEntity metaTileEntity, FuelRecipeMap recipeMap, S
     }
 
     public long getRecipeOutputVoltage() {
-        return recipeOutputVoltage;
+        return calculateRecipeOutputVoltage();
```

Comment:
Do you understand the difference between getXXX and calculateXXX?
getXXX methods are supposed to return cached values
calculateXXX methods are used to actually calculate values and then cache them into the fields

So why do you call calculateRecipeOutputVoltage inside of getRecipeOutputVoltage

Reply:
simple yet it requires to read every comment this is to fix  the `FueledMultiblockController` problem caused by the large turbine overriding this methods. If you find a better way to handle possible modification of this methods not breaking what is showed in the GUI go ahead and tell this in comment.

Reply:
large turbine overrides this method because it was written... before that PR?
With code introduced in this PR, getters should be literally made final, and large turbine should override calculateRecipeOutputVoltage

Reply:
yes but that may break code which use the same behavior in addon this is why I don't touch at any other class to fix issue in this class this way I am sure I won't be obviously breaking things

Reply:
We literally have all addon developers in touch with GTCE changes, doing changes like that is not an issue at all.

Reply:
The point of calling it getX() instead of calculateX() is to differentiate policy from metrics.

It is very confusing to have:

calculateSomething(Recipe r) // override this to change behaviour
calculateSomething() { return calculateSomething(lastRecipe)) // doesn't change behaviour, just exposes info for display

If you believe getX() should be reserved for javabean accessors then some other prefix should be used that makes it obvious it is not a policy override

But the rational for the extra dance in this location is backwards compatibility:
https://github.com/GregTechCE/GregTech/pull/1604#issuecomment-846636440
This is to workaround the hack where LargeTurbine changes the contract of getRecipeOutputVoltage() to be every tick instead of at the start of the recipe and the possiblity that an addon has overridden it.

Reply:
>We literally have all addon developers in touch with GTCE changes, doing changes like that is not an issue at all.

LAG Idiot 12 day ago:
>We should try not break our addons even if they are using common part of mod.

when talking about a small and easy to fix changes in the common part of GTCE . here we are in the API I rather have a method with a name a little weird when looking at what it does than breaking things in the API.

Reply:
the point of calling it getX is to indicate that it retrieves metrics, indeed.
problem is, in that exact case it does NOT retrieve metrics, it literally recomputes value in place instead of providing value that was ACTUALLY used this tick for adding energy into the buffer.

getX methods usually do not indicate a policy to override, even if they are not marked as final. But yeah, to avoid confusion, you can literally mark them final to prevent people from overriding them instead of calculateX methods.

Again, like I said above, we have literally all addon developers in touch with GTCE development, so even if it breaks any functionality in addons, fix should not take much time to implement.

Reply:
> when talking about a small and easy to fix changes in the common part of GTCE . here we are in the API I rather have a method with a name a little weird when looking at what it does than breaking things in the API.
If you really want to keep compatibility that much, you should actually deprecate this method and provide a proper getter method which does not recompute value

Reply:
>If you really want to keep compatibility that much, you should actually deprecate this method and provide a proper getter method which does not recompute value

in deed I should, I didn't really think about this.

> it literally recomputes value in place instead of providing value that was ACTUALLY used this tick for adding energy into the buffer.

No need for caps locks.

yes values used to compute could have changed depending on the calling order of block updates this should get fixed

Reply:
>If you really want to keep compatibility that much, you should actually deprecate this method and provide a proper getter method which does not recompute value

just think about that but is it really needed to replace this method with a new one ? is there a reason why you would need the original recipe duration and not the effective one ?